src/Core/Util/ConfigurationVariablesLoader.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Util;
+
+use Symfony\Component\Dotenv\Dotenv;
+
+class ConfigurationVariablesLoader
+{
+    private const PARAMETERS_WITH_ENV_VARS = [
+        'database_host' => 'PS_DATABASE_HOST',
+        'database_port' => 'PS_DATABASE_PORT',
+        'database_user' => 'PS_DATABASE_USER',
+        'database_password' => 'PS_DATABASE_PASSWORD',
+        'database_name' => 'PS_DATABASE_NAME',
+        'database_prefix' => 'PS_DATABASE_PREFIX',
+        'database_engine' => 'PS_DATABASE_ENGINE',
+        'cookie_key' => 'PS_COOKIE_KEY',
+        'cookie_iv' => 'PS_COOKIE_IV',
+        'new_cookie_key' => 'PS_NEW_COOKIE_KEY',
+        'mailer_transport' => 'PS_MAILER_TRANSPORT',
+        'mailer_host' => 'PS_MAILER_HOST',
+        'mailer_user' => 'PS_MAILER_USER',
+        'mailer_password' => 'PS_MAILER_PASSWORD',
+        'locale' => 'PS_LOCALE',
+        'secret' => 'PS_SECRET',
+        'ps_caching' => 'PS_CACHING',
+        'ps_cache_enable' => 'PS_CACHE_ENABLE',
+        'ps_creation_date' => 'PS_CREATION_DATE',
+        'use_debug_toolbar' => 'PS_USE_DEBUG_TOOLBAR',
+    ];
+
+    public function __construct(string $envFilePath)
+    {
+        (new Dotenv(true))->loadEnv($envFilePath);
+    }
+
+    public function loadEnvVariables(array $parameters): array
```

Comment:
```suggestion
    public function loadEnvironmentVariables(array $parameters): array
```

What do you think? Do we use full name or short name?

Reply:
I'll change this. With auto completion in most of IDE, sounds better to have full names.

Reply:
Agreed 👍 

Also short names like "env", "conf", "Mgr" can be confusing especially for junior developers